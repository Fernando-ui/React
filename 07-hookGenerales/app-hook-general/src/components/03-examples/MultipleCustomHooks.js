import React from 'react'
import { useCounter } from '../../hooks/useCounter';
import { useFetch } from '../../hooks/useFetch'
import '../02-useEffect/effects.css'


export const MultipleCustomHooks = () => {

    const { counter, increment, decrement} = useCounter();
    const {loading, data} = useFetch(`https://www.breakingbadapi.com/api/quotes/${counter}`);

    const { author, quote } = !!data && data[0];
    console.log(author, quote);
    
    
    
    return (
        <>
            <h1>Custom Hooks</h1>
            <h2>Breaking Bad</h2>
            
            {
                loading
                ?
                (
                    <div className="alert alert-info text-center">
                        Loading...
                    </div>

                )
                :
                (
                    <blockquote className="blockquote text-right">

                        <p className="mb-0 texto-parrafo">{quote}</p>
                        <footer className="blockquote-footer footer">{author}</footer>

                    </blockquote>

                )
            }
            <div className="container-Botones">
            <button onClick={ decrement } className="btn btn-primary boton-anterior">Anterior</button>
            <button onClick={ increment } className="btn btn-primary boton-siguiente">Siguiente</button>

            </div>

        </>
    )
}
