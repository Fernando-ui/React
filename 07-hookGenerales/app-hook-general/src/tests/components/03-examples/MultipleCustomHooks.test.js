import React from 'react';
import { shallow } from 'enzyme';
import { MultipleCustomHooks } from '../../../components/03-examples/MultipleCustomHooks';

describe('Pruebas en <MultipleCustomHooks/>', () => {
    
    test('Debe de mostrarse correctamente', () => {
   
        
        const wrapper = shallow(<MultipleCustomHooks />);
        expect(wrapper ).toMatchSnapshot();


    });
    

});
