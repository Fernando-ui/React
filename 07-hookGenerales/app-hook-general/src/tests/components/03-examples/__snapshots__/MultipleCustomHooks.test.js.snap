// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <MultipleCustomHooks/> Debe de mostrarse correctamente 1`] = `
<Fragment>
  <h1>
    Custom Hooks
  </h1>
  <h2>
    Breaking Bad
  </h2>
  <div
    className="alert alert-info text-center"
  >
    Loading...
  </div>
  <div
    className="container-Botones"
  >
    <button
      className="btn btn-primary boton-anterior"
      onClick={[Function]}
    >
      Anterior
    </button>
    <button
      className="btn btn-primary boton-siguiente"
      onClick={[Function]}
    >
      Siguiente
    </button>
  </div>
</Fragment>
`;
